addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1807,6 +1807,240 @@ const SelectPagerUserValueWidget = SelectUserValueWidget.extend({
+            fields: '[]',
+            domain: '[]',
+            callWith: 'id',
+        });
+        this.configAttributes.forEach(attr => {
+            if (dataAttributes.hasOwnProperty(attr)) {
+                options[attr] = dataAttributes[attr];
+                delete dataAttributes[attr];
+            }
+        });
+        options.limit = parseInt(options.limit);
+        options.fields = JSON.parse(options.fields);
+        if (!options.fields.includes('display_name')) {
+            options.fields.push('display_name');
+        }
+        options.domain = JSON.parse(options.domain);
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+
+        this.inputEl = document.createElement('input');
+        this.inputEl.setAttribute('placeholder', _t("Search for records..."));
+        const searchEl = document.createElement('div');
+        searchEl.classList.add('o_we_m2o_search');
+        searchEl.appendChild(this.inputEl);
+        this.menuEl.appendChild(searchEl);
+
+        this.searchMore = document.createElement('div');
+        this.searchMore.classList.add('o_we_m2o_search_more');
+        this.searchMore.textContent = _t("Search more...");
+        this.searchMore.title = _t("Search to show more records");
+        return this._search('');
+    },
+    /**
+     * @override
+     */
+    async setValue(value, methodName) {
+        await this._super(...arguments);
+        // The currently selected value is not present in the search, need to read
+        // its display name.
+        if (this.menuTogglerEl.textContent === '/' && value !== '') {
+            this.menuTogglerEl.textContent = await this._getDisplayName(parseInt(value));
+        }
+    },
+    /**
+     * Prevents double widget instanciation for we-buttons that have been
+     * created manually by _search (container widgets will have their innner
+     * html searched for userValueWidgets to instanciate during option startup)
+     *
+     * @override
+     */
+    isContainer() {
+        return false;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Searches the database for corresponding records and updates the dropdown
+     *
+     * @private
+     */
+    async _search(needle) {
+        const recTuples = await this._rpc({
+            model: this.options.model,
+            method: 'name_search',
+            kwargs: {
+                name: needle,
+                args: this.options.domain,
+                operator: "ilike",
+                limit: this.options.limit + 1,
+            },
+        });
+        const records = await this._rpc({
+            model: this.options.model,
+            method: 'read',
+            args: [recTuples.map(([id, name]) => id), this.options.fields],
```

Comment:
```suggestion
            args: [recTuples.map(([id, _]) => id), this.options.fields],
```

Reply:
Or `_name` if you prefer.

Reply:
At that point I might as well just not destructure it out of the argument and only use `[id]`. I tend to avoid using `_` as an ignored argument name because it shadows underscorejs.

Reply:
> At that point I might as well just not destructure it out of the argument and only use [id].

Sure. I'm not used to languages who just... let you do that. In Python it's going to raise a typeerror.

> I tend to avoid using _ as an ignored argument name because it shadows underscorejs.

Hence the suggestion of just prefixing with an underscore. Point is to avoid "unused variable" warnings as not all linters will ignore unused parameters.